src/content/reference/react-dom/components/input.md

```diff
@@ -456,140 +457,135 @@ function SignupForm() {
+Kodunuzu düzeltmek için, senkron bir şekilde `e.target.value` değeriyle güncelleyiniz:
 
 ```js
 function handleChange(e) {
-  // ✅ Updating a controlled input to e.target.value synchronously
+  // ✅ Kontrollü girdiyi e.target.value değeriyle senkron güncelleme
   setFirstName(e.target.value);
 }
 ```
 
-If this doesn't fix the problem, it's possible that the input gets removed and re-added from the DOM on every keystroke. This can happen if you're accidentally [resetting state](/learn/preserving-and-resetting-state) on every re-render, for example if the input or one of its parents always receives a different `key` attribute, or if you nest component function definitions (which is not supported and causes the "inner" component to always be considered a different tree).
-
+Eğerbunlar sorununuzu çözmezse, girdiniz her klavye girişinde DOM'dan silinip geri ekleniyor olabilir. Bu durum, her yeniden renderda [durumu sıfırlama](/learn/preserving-and-resetting-state)'dan kaynaklı olabilir. Örneğin, eğer girdiniz ya da onun üstündeki elemanlardan biri süreki farklı `key` niteliği alıyor olabilir veya bileşen tanımlarını iç içe yerleştirmenizden kaynaklı olabilir (iç içe yerleştirmeler desteklenmemektedir ve "iç" bileşenin her zaman farklı bir ağaç olarak kabul edilmesine neden olur).
 ---
 
-### I'm getting an error: "A component is changing an uncontrolled input to be controlled" {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}
-
+### "Bir bileşen kontrolsüz girişi kontrol ediyor" hatası alıyorum  {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}
 
-If you provide a `value` to the component, it must remain a string throughout its lifetime.
 
-You cannot pass `value={undefined}` first and later pass `value="some string"` because React won't know whether you want the component to be uncontrolled or controlled. A controlled component should always receive a string `value`, not `null` or `undefined`.
+Bileşeninize `value` değeri verdiyseniz, yaşam döngüsü boyunca string olarak kalmalıdır.
 
-If your `value` is coming from an API or a state variable, it might be initialized to `null` or `undefined`. In that case, either set it to an empty string (`''`) initially, or pass `value={someValue ?? ''}` to ensure `value` is a string.
+Önce `value={undefined}` olarak değer iletip daha sonra `value="some string"` şeklinde değer iletemezsiniz çünkü React, kontrollü bileşen mi kontrolsüz bileşen mi bunu bilemez. Kontrollü bileşen her zaman `value` değer olarak string almalıdır, `null` ya da `undefined` almamalıdır.
 
-Similarly, if you pass `checked` to a checkbox, ensure it's always a boolean.
+`value` değeriniz API'den veya durum değişkeninden geliyorsa, başta `null` veya `undefined` olarak tanımlanmış olabilir. Bu durumda boş bir string (`''`) olarak tanımlayabilirsiniz veya `value={someValue == ''}` olarak iletip `value` değerinin string olup olmadığına emin olabilirsiniz.
```

Comment:
Buradaki `someValue ?? ''` kodunu değiştirmişsiniz sanırım. `someValue == ''` doğru şekilde çalışmaz. 

Yanlışlıkla öyle yazıldığını düşündünüz sanırım. `??` geçerli bir operator. (Nullish coalescing operator). 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

Reply:
Öncelikle geri dönütünüz için teşekkür ederim. Burada ?? yerine yanlışlıkla == kullanmışım, diğer yorumunuzla birlikte burayı da hemen düzeltiyorum.